src/doom/g_game.c

```diff
@@ -2541,7 +2541,7 @@ void G_WriteDemoTiccmd (ticcmd_t* cmd)
     byte *demo_start;
 
     if (gamekeydown[key_demo_quit])           // press q to end demo recording 
-	G_CheckDemoStatus (); 
+	G_CheckDemoStatus (cmd); 
```

Comment:
I think everything would still work if this was `NULL`, but I figured I would include `cmd` in any context that had it for consistency.

Reply:
But then you would record a "joined" marker in the last tic even if the demo was quit normally.

Reply:
The joined marker will only be placed if `demoplayback` and `demorecording` are both true when this method is called, so I don't believe that can happen in normal demo recording.

Reply:
Ah, okay, sure. So, apparently, I find this part a bit confusing. 😉